File: src/config/platform_syntax.py
"""
  This module contains an object that maps different platforms to their respective link syntax.
  Use this to help the AI understand the markdown discrepancies between platforms.
"""
platform_syntax = {
    "Slack": {
        "links": "<https://example.com|link description/text>",
        "bold": "*bold text* - DO NOT USE TWO ASTERISKS SLACK WILL NOT RENDER IT",
        "italic": "_italic text_",
        "code": "`code`",
        "code_block": "```\ncode block\n```",
        "list": "- item 1\n- item 2",
        "numbered_list": "1. item 1\n2. item 2",
        "quote": "> This is a quote",
        "strikethrough": "~strikethrough~",
    },
    "Discord": {
        "links": "[link description/text](https://example.com)",
        "bold": "**bold text**",
        "italic": "_italic text_",
        "code": "`code`",
        "code_block": "```\ncode block\n```",
        "list": "- item 1\n- item 2",
        "numbered_list": "1. item 1\n2. item 2",
        "quote": "> This is a quote",
        "strikethrough": "~~strikethrough~~",
        "underline": "__underline__",
        "spoiler": "||spoiler||",
        "header": "# Top Level Header\n## Second Level Header\n### Third Level Header",
    }
}

File: src/config/platformSyntax.ts
/**
 * @copyright Deepgram
 * @license MIT
 * @author Naomi Carrigan
 */

export const platformSyntax = {
  discord: {
    block:         "```lang\ncode block\n```",
    bold:          "**bold**",
    code:          "`code`",
    header:        "# H1, ## H2, ### H3",
    italic:        "_italic_",
    link:          "[link](https://url)",
    list:          "- item 1\n- item 2",
    numberedList:  "1. item 1\n2. item 2",
    quote:         "> quote",
    spoiler:       "||spoiler||",
    strikethrough: "~~strikethrough~~",
    underline:     "__underline__",
  },
  slack: {
    block:         "```\ncode block\n```",
    bold:          "*bold*",
    code:          "`code`",
    italic:        "_italic_",
    link:          "<https://url|link>",
    list:          "- item 1\n- item 2",
    numberedList:  "1. item 1\n2. item 2",
    quote:         "> quote",
    strikethrough: "~strikethrough~",
  },
};
